test/forwarder/BaseRelayRecipient.test.ts

```diff
@@ -56,4 +56,17 @@ contract('BaseRelayRecipient', ([from]) => {
     assert.equal(await callMsgData(fwd.address, extra + sender.slice(2)), encoded + extra,
       'should extract msg.data if called through trusted forwarder')
   })
+
+  it('should extract msgSender and msgData in transaction', async () => {
+    // trust "from" as forwarder (using real forwarder requires signing
+    const recipient = await TestForwarderTarget.new(from)
+    const encoded = recipient.contract.methods.emitMessage('hello').encodeABI() as string
+    const encodedWithSender = encoded + '1'.repeat(40)
```

Comment:
oh just use some address, you have 10 of those for free

what if it like gave you address in reverse? would not catch it with 0x111111111111111111111111111111111111111111, you know...

Reply:
ok